### PR TITLE
PEER-626 Add page with moderator rights

### PR DESCRIPTION
**Pull request by almondy22**
Added page with permissions for moderators.

### CODE REVIEW

File: app/containers/Moderation/Content.jsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  :hover {
+    cursor: pointer;
+  }
+
+  @media only screen and (max-width: 576px) {
+    margin-right: 8px;
+  }
+`;
+
+const QuestionBox = BaseTransparent.extend`
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:80]: Name `Question` for components does not make sense. Please give better names.

File: app/containers/Moderation/Content.jsx [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+`.withComponent('li');
+
+const QuestionBoxBody = styled.div`
+  width: 100%;
+`;
+
+const Question = ({
+  description,
+  title,
+  questionCode,
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:108]: Fix argument names to makes sense

File: app/containers/Moderation/Content.jsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  sectionCode,
+  getQuestionCode,
+  permission,
+}) => {
+  const questionId = getQuestionCode(sectionCode, questionCode);
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:113]: Fix variables names to make sense

File: app/containers/Moderation/Content.jsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    </QuestionBox>
+  );
+};
+
+const Section = ({
+  h2,
+  blocks,
+  sectionCode,
+  route,
+  getSectionCode,
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:143]: Fix names

File: app/containers/Moderation/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import * as routes from 'routes-config';
+import { FormattedMessage } from 'react-intl';
+
+import commonMessages from 'common-messages';
+
+import { TEXT_PRIMARY } from 'style-constants';
+import faqPageHeader from 'images/faqPageHeader.svg?inline';
```
Comment (sergeyi23) [inline app/containers/Moderation/Header.jsx:8]: Need to use some other icon here. Not FAQ.

File: app/utils/properties.js [hunk 6]
```diff
@@ -18,6 +20,45 @@ const findAllPropertiesByKeys = (properties, keys) =>
...
 
+export const getModeratorPermissions = (
+  communityPermissions = [],
+  globalPermissions = [],
+  isGlobal,
+  communities,
+) => {
+  const values = isGlobal ? globalPermissions : communityPermissions;
+  const perms = isGlobal ? moderatorPermissions : communityAdminPermissions;
+  const permissions = { h1: 'Moderator permissions' };
```
Comment (sergeyi23) [inline app/utils/properties.js:31]: Translation is required for this text.

File: app/utils/properties.js [hunk 7]
```diff
@@ -18,6 +20,45 @@ const findAllPropertiesByKeys = (properties, keys) =>
...
+      .toString(2)
+      .split('')
+      .forEach((perm, permIndex) => {
+        if (perm === '1') permission.push(permIndex);
+      });
+    return [
+      ...acc,
+      {
+        h2: isGlobal
+          ? 'Global Moderator'
```
Comment (sergeyi23) [inline app/utils/properties.js:44]: Translation is required for this text.

**Review by sergeyi23 — CHANGES_REQUESTED** [review]
Please see the comments.

File: app/containers/Moderation/Content.jsx [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+const Permission = ({
+  description,
+  title,
+  permissionCode,
+  sectionCode,
+  getPermissionCode,
+  permission,
+}) => {
+  const questionId = getPermissionCode(sectionCode, permissionCode);
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:113]: Give better name to variable `questionId`

File: app/containers/Moderation/Content.jsx [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+  let ico;
+  if (permission.includes(permissionCode)) {
```
Comment (sergeyi23) [inline app/containers/Moderation/Content.jsx:115]: curly braces are inconsistent. If use them in if statement then should use them for else as well.

Comment (sergeyi23) [reply app/containers/Moderation/Content.jsx:115]: That assignment can be simplified with the conditional:
`const ico = permission.includes(permissionCode) ? okayGreen  : notOkay ;`

File: app/containers/Moderation/Header.jsx [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import * as routes from 'routes-config';
+import { FormattedMessage } from 'react-intl';
+
+import commonMessages from 'common-messages';
+
+import { TEXT_PRIMARY } from 'style-constants';
+import faqPageHeader from 'images/incognito.svg?inline';
```
Comment (sergeyi23) [inline app/containers/Moderation/Header.jsx:8]: `faqPageHeader` should be renamed

File: app/images/incognito.svg [hunk 11]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="32px" height="32px" viewBox="0 0 32 32"><g stroke-width="2" transform="translate(0, 0)"><polygon points="3 16 29 16 25 11 7 11 3 16" fill="none" stroke="#444444" stroke-miterlimit="10" stroke-width="2" data-cap="butt" stroke-linecap="butt" stroke-linejoin="miter"></polygon><polyline points="7 11 9 2 23 2 25 11" fill="none" stroke="#444444" stroke-miterlimit="10" stroke-width="2" data-cap="butt" stroke-linecap="butt" stroke-linejoin="miter"></polyline><circle cx="24" cy="25" r="5" fill="none" stroke="#444444" stroke-miterlimit="10" stroke-width="2" data-cap="butt" data-color="color-2" stroke-linecap="butt" stroke-linejoin="miter"></circle><circle cx="8" cy="25" r="5" fill="none" stroke="#444444" stroke-miterlimit="10" stroke-width="2" data-cap="butt" data-color="color-2" stroke-linecap="butt" stroke-linejoin="miter"></circle><path d="M19.32,23.287a4.926,4.926,0,0,0-6.64,0" fill="none" stroke="#444444" stroke-miterlimit="10" stroke-width="2" data-cap="butt" data-color="color-2" stroke-linecap="butt" stroke-linejoin="miter"></path></g></svg>
```
Comment (sergeyi23) [inline app/images/incognito.svg:1]: Did you get this icon from our icons collection?

File: app/common-messages.js [hunk 12]
```diff
@@ -33,6 +33,54 @@ export default defineMessages({
   achievements: {
     id: 'app.common.achievements',
   },
+  moderation: {
```
Comment (sergeyi23) [inline app/common-messages.js:36]: common-messages.js should be used only for messages that are used in more than one component / container. These messages should moved to local messages in the new component.

**Review by sergeyi23 — CHANGES_REQUESTED** [review]
Please see several new comments

File: app/utils/properties.js [hunk 13]
```diff
@@ -18,6 +21,46 @@ const findAllPropertiesByKeys = (properties, keys) =>
...
+  const values = isGlobal ? globalPermissions : communityPermissions;
+  const perms = isGlobal ? moderatorPermissions : communityAdminPermissions;
+  const permissions = {};
+  permissions.blocks = values.reduce((acc, { community, value }, index) => {
+    const permission = [];
+    value
+      .toString(2)
+      .split('')
+      .forEach((perm, permIndex) => {
+        if (perm === '1') permission.push(permIndex);
```
Comment (sergeyi23) [inline app/utils/properties.js:40]: what is perm === '1'? Looks like need some descriptive constant for value '1'

File: app/images/incognito.svg (hunk 11 above)
Comment (almondy22) [reply app/images/incognito.svg:1]: Yes, from nucleoapp

**Review by sergeyi23 — APPROVED** [review]

**Pull request merged**